ports/geckolib/gecko_bindings/ptr.rs

```diff
@@ -0,0 +1,64 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use bindings::*;
+use heapsize::HeapSizeOf;
+use std::fmt::{self, Debug};
+
+macro_rules! define_holder_gecko_arc {
+    ($x:ident, $y:ident, $z:ident) => (
+        #[derive(PartialEq)]
+        pub struct $x {
+            ptr: *mut $z,
+        }
+
+        impl $x {
+            pub fn new(data: *mut $z) -> $x {
+                assert!(!data.is_null());
+                unsafe { concat_idents!(Gecko_AddRef, $y, ArbitraryThread)(data); }
+                $x {
+                    ptr: data
+                }
+            }
+
+            pub fn as_raw(&self) -> *mut $z { self.ptr }
+        }
+
+        unsafe impl Send for $x {}
+        unsafe impl Sync for $x {}
+
+        impl Clone for $x {
+            fn clone(&self) -> $x {
+                $x::new(self.ptr)
+            }
+
+            fn clone_from(&mut self, source: &$x) {
+                unsafe {
```

Comment:
I think we can just take the default impl and not override this here - we don't save any work unless I'm missing something.